docgen/middlewares.js

```diff
@@ -102,6 +103,20 @@ const common = [
   }),
   // since we use @import, autoprefixer is used after sass
   autoprefixer,
+  copy({
```

Comment:
Can you explain this part, @vvo ? Why do we need to move files to the v2 folder now?

Reply:
Those two copy calls turns:
- docs/index.html
- docs/*

into
- docs/v2/*
- docs/index.html

Which is what we want to have in production too.

So that this is exactly the setup that will be used in every env (dev, preview, production). Leaving few differences between those envs, ensuring that if we use paths like a href etc then dev, preview or production will behave the same.

We could even move further and even in dev have /instantsearch.js/v2 to remove some env differences.

Ultimately we then push the docs/ folder to gh-pages.

Other possibilities would have been to only modify the gh-pages script and do the moves there but that would have been a bit too magical I felt and error prone.

Let me know if you want to change it.

Reply:
The iso env seems appealing but it's gonna be less practical (how someone know that they should go to netlify/instantsearch.js when opening a PR). That's a good as of now I think, closer to what we want 👍

Reply:
It automatically redirects on netlify :)

Reply:
Oh wait not yet, I will add that

Reply:
There's no need for any redirection, default netlify preview is the V2 homepage which is what we want.